### PR TITLE
[mds-core] Bump dependencies and resolve new typescript compilation/linter complaints

### PR DESCRIPTION
**Pull request by avatarneil**
## PR Checklist

 - [x] simple searchable title - `[mds-db] Add PG env var`, `[config] Fix eslint config`
 - [x] briefly describe the changes in this PR

## Impacts
- [x] Provider
- [x] Agency
- [x] Audit
- [x] Policy
- [x] Compliance
- [x] Daily
- [x] Native
- [x] Policy Author

### CODE REVIEW

File: packages/mds-agency/api.ts [hunk 1]
```diff
@@ -444,7 +444,7 @@ function api(app: express.Express): express.Express {
     vehicles: (Device & { updated?: number | null; telemetry?: Telemetry | null })[]
   }> {
     function fmt(query: { skip: number; take: number }): string {
-      const flat = Object.assign({}, reqQuery, query)
+      const flat = { ...reqQuery, ...query }
```
Comment (karcass) [inline packages/mds-agency/api.ts]: <3

File: packages/mds-db/index.ts [hunk 2]
```diff
@@ -1377,7 +1378,7 @@ async function readRule(rule_id: UUID): Promise<Rule> {
 
 async function readUnprocessedStatusChangeEvents(
   before: Recorded<VehicleEvent> | null,
-  take: number = 1000
+  take = 1000
```
Comment (karcass) [inline packages/mds-db/index.ts]: number was removed because it's irrelevant?

**Comment by karcass** [comment]
👍

File: packages/mds-agency/api.ts (hunk 1 above)
Comment (wesleyfsmith) [reply packages/mds-agency/api.ts]: love me some spread operator

**Pull request merged**